### PR TITLE
Handle infinity values in Tensor.where

### PR DESCRIPTION
**Pull request by jla524**
for #942 (cc: @dc-dc-dc)
It doesn't handle nan yet, requires #1069 to be resolved.

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         2               2
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -167,8 +167,8 @@ def ones_like(tensor, **kwargs): return Tensor.full_like(tensor, 1, **kwargs)
   def eye(dim, **kwargs): return Tensor([1], **kwargs).slice(((0,dim+1),)).reshape(1, dim+1).expand(dim, dim+1).reshape(dim*(dim+1)).slice(((0,dim*dim),)).reshape(dim, dim)
 
   def where(self:Tensor, input_:Union[Tensor, float], other:Union[Tensor, float]):
-    cond = (self != 0.0)
-    return cond * input_ + (1.0 - cond) * other
+    cond, input_inf, other_inf = (self != 0.0), (input_ == float("inf")), (other == float("inf"))
+    return cond * (input_ ** (1 - input_inf)) + (1.0 - cond) * (other ** (1 - other_inf))
```
Comment (geohot) [inline tinygrad/tensor.py:171]: Do you have to use pow?

Comment (jla524) [reply tinygrad/tensor.py]: Nope, we can implement this without pow.

Comment (jla524) [reply tinygrad/tensor.py]: Sorry, this PR wasn't ready for review. I'm not sure if using == or != is the right approach. We don't have the backward pass implemented.

**Pull request closed without merging**